### ISSUE TITLE
3.4-RC1.1 Project config sync does not create multiple entry types

### ISSUE THREAD

**Issue by nicgordon** [craftcms/cms#5446]
### Description

If you have a structure section in your project config which has multiple entry types nested within it, a project config sync will only create one of them. It will create the first one in the collection sorted by the random uuid that gets assigned to it in the project config.

Manually running `./craft project-config/sync` also does not resolve the issue.

I have tested on Craft v3.3.15 and it does not happen there.

### Steps to reproduce

1. Add the project config below to a fresh install of Craft 3.4
2. Go through the install process (`/admin/install`)
3. Go to Settings > Sections and view the Entry Types for the section called `Test`. There should be 4, but there is only 1.

### Additional info

- Craft version: 3.4-RC1.1
- PHP version: 7.3.11
- Database driver & version: PostgreSQL 11.5 and MySQL 5.7
- Plugins & versions: None

### Project config to test with

```
dateModified: 1579156830
email:
  fromEmail: test@gmail.com
  fromName: Test
  transportType: craft\mail\transportadapters\Sendmail
fieldGroups:
  37883cbd-1fc3-4026-9dc7-fb6eaffb4fd8:
    name: Common
sections:
  d97da60e-5036-4759-b94f-7b2f5486393c:
    enableVersioning: true
    entryTypes:
      04b3f349-8540-4c41-80fb-6a7ed2f918d2:
        handle: nested4
        hasTitleField: true
        name: Nested4
        sortOrder: 4
        titleFormat: ''
        titleLabel: Title
      10507f1e-b165-4dd2-b82a-f9573a25561f:
        handle: nested2
        hasTitleField: true
        name: Nested2
        sortOrder: 2
        titleFormat: ''
        titleLabel: Title
      3c401871-5678-4c7d-8a4d-a3cee58af945:
        handle: nested3
        hasTitleField: true
        name: Nested3
        sortOrder: 3
        titleFormat: ''
        titleLabel: Title
      44ca81e7-300e-41ad-89ae-0ac9bfc19a97:
        handle: nested1
        hasTitleField: true
        name: Nested1
        sortOrder: 1
        titleFormat: ''
        titleLabel: Title
    handle: test
    name: Test
    propagationMethod: all
    siteSettings:
      5ecf171b-a315-47f0-8b9a-e708105704a2:
        enabledByDefault: true
        hasUrls: true
        template: ''
        uriFormat: 'test/{slug}'
    structure:
      maxLevels: null
      uid: fabd615c-a883-4d31-ad7a-f2243db8f61f
    type: structure
siteGroups:
  71b6c696-161d-4360-83f4-08d7127c00a8:
    name: Test
sites:
  5ecf171b-a315-47f0-8b9a-e708105704a2:
    baseUrl: $DEFAULT_SITE_URL
    handle: default
    hasUrls: true
    language: en-AU
    name: Test
    primary: true
    siteGroup: 71b6c696-161d-4360-83f4-08d7127c00a8
    sortOrder: 1
system:
  edition: solo
  live: true
  name: Test
  schemaVersion: 3.4.6
  timeZone: America/Los_Angeles
users:
  allowPublicRegistration: false
  defaultGroup: null
  photoSubpath: ''
  photoVolumeUid: null
  requireEmailVerification: true
```

**Comment by brandonkelly** [comment]
We just released Craft 3.4 RC2 with a fix for this. Thanks for reporting!

**Comment by nicgordon** [comment]
Brilliant, thank you!
